ipynb/examples/utils/gem5_example.ipynb

```diff
@@ -0,0 +1,570 @@
+    "# ~15 minutes to start from scratch (atomic)\n",
+    "te = TestEnv(conf)\n",
+    "target = te.target"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "deletable": true,
+    "editable": true
+   },
+   "source": [
+    "# Run workloads on gem5"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This is an example of running a workload and extracting stats from the simulation using m5 commands. For more information about m5 commands, see http://gem5.org/M5ops"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "collapsed": false,
+    "deletable": true,
+    "editable": true,
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2017-05-23 22:19:49,308 DEBUG   : ssh          : Pushing /home/vagrant/tools/arm64/sysbench to device.\n",
+      "2017-05-23 22:19:49,309 DEBUG   : ssh          : gem5interactdir: /tmp/wa_4\n",
+      "2017-05-23 22:19:49,310 DEBUG   : ssh          : dest: /home/root/devlib-target/bin/sysbench\n",
+      "2017-05-23 22:19:49,310 DEBUG   : ssh          : filename: sysbench\n",
+      "2017-05-23 22:20:02,329 DEBUG   : ssh          : Push complete.\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "''"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Initialise command: [binary/script, arguments]\n",
+    "command = 'sysbench'\n",
+    "args = '--test=cpu --max-time=1 run'\n",
+    "\n",
+    "# Install binary if needed\n",
+    "path = target.install_if_needed(\"/home/vagrant/tools/arm64/\" + command)\n",
+    "\n",
+    "# Reset and dump statistics\n",
+    "target.execute('m5 resetstats')\n",
+    "\n",
+    "# FTrace the execution of this workload\n",
+    "te.ftrace.start()\n",
+    "target.execute(path + \" \" + args)\n",
+    "te.ftrace.stop()\n",
+    "\n",
+    "target.execute('m5 dumpstats')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "deletable": true,
+    "editable": true
+   },
+   "source": [
+    "# gem5 stats analysis\n",
+    "In the previous section, we used m5 to dump stats about the platform that is being simulated by gem5. These stats are pulled from the virtio directory and placed in the LISA results folder, but they require some parsing which is done here"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "collapsed": false,
+    "deletable": true,
+    "editable": true,
+    "scrolled": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n",
+      "After sysbench : \n",
+      "-------------------\n",
+      "Seconds in host --> 0.96\n",
+      "Seconds in guest --> 0.021238\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Measure the run times\n",
```

Comment:
This is another great example of commands which should be encapsulated into an API, either in devlib or in LISA in this case (since it's about data extraction and presentation).
It would be probably worth to generate and return a DataFrame out of these data.